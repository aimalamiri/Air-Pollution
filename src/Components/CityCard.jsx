import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const CityCard = (props) => {
  const { city } = props;
  return (
    <Link to={`/${city.Name}`}>
      <div className="p-2 shadow bg-gradient-to-br from-blue-400 to-indigo-700 min-h-full">
        <div className="flex justify-center">
          <img
            src={`https://countryflagsapi.com/svg/${city.Country}`}
            className="w-1/2 opacity-20"
            alt={`${city.Country} flag`}
          />
        </div>
        <div className="flex flex-wrap justify-end mt-5 font-bold text-sm text-white">
          <div>
            <h3 className="uppercase">{city.Name}</h3>
            <div className="">
              <h5>{city.Population.toLocaleString()}</h5>
            </div>
          </div>
        </div>
      </div>
    </Link>
  );
};

CityCard.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  city: PropTypes.object.isRequired,
};

export default CityCard;
