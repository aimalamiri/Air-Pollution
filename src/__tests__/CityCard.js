import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CityCard from '../Components/CityCard';

it('Renders the CityCard component', () => {
  const city = {
    Name: 'Kabul',
    Population: 7000000,
  };
  const tree = renderer
    .create(
      <MemoryRouter>
        <CityCard city={city} />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
