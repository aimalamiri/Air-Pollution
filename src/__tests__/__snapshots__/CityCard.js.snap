// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the CityCard component 1`] = `
<a
  href="/Kabul"
  onClick={[Function]}
>
  <div
    className="p-2 shadow bg-gradient-to-br from-blue-400 to-indigo-700 min-h-full"
  >
    <div
      className="flex justify-center"
    >
      <img
        alt="undefined flag"
        className="w-1/2 opacity-20"
        src="https://countryflagsapi.com/svg/undefined"
      />
    </div>
    <div
      className="flex flex-wrap justify-end mt-5 font-bold text-sm text-white"
    >
      <div>
        <h3
          className="uppercase"
        >
          Kabul
        </h3>
        <div
          className=""
        >
          <h5>
            7,000,000
          </h5>
        </div>
      </div>
    </div>
  </div>
</a>
`;
